app/scripts/controllers/transactions/index.js

```diff
@@ -726,6 +707,7 @@ export default class TransactionController extends EventEmitter {
       };
       this.txStateManager.setTxStatusConfirmed(txId);
       this._markNonceDuplicatesDropped(txId);
+      this._trackTxEvent(txMeta, 'Transaction Finalized');
```

Comment:
Let's pull some values out of the txReceipt such as gasUsed, and explore in that object because I'm pretty sure we can get the time it was included in the network. We don't want to include that exact time, but instead txMeta.submittedTime - txReceipt contains blockNumber and then you can use getBlockByNumber to get the timestamp of when the block was collated.

Reply:
without going the extra mile of solving #9419, we should at least catch cases wher the txReceipt.status is '0x0' which is onChain, and track that case as a failure for metrics instead of as a success. We also track a metrics event for on chain failures in metamask-controller (check for looking at txReceipt.status ,and if you search for txReceipt?.status you'll find a few more cases of checking that in the UI). 

I think if we can just check the status and log the error as an event here we have fulfilled our duty to this story and we can revisit #9419 and the differences between on-chain and in-metamask failures later ;)

Reply:
`completion_time` is now factored in and included in the metrics payload here